### DISCUSSION TITLE
use turborepo

### DISCUSSION THREAD

**Discussion by threepointone** [cloudflare/workers-sdk#1990] [General]
Would turborepo make our builds faster? Would be interesting to try

**Comment by petebacondarwin** [comment]
"Monorepos that make ship happen." LOL!

**Comment by petebacondarwin** [comment]
That looks interesting...

**Comment by GregBrimble** [comment]
https://nx.dev/ is another option

**Comment by petebacondarwin** [comment]
Yes!

But let's make sure we avoid https://bazel.build/ for now! (Not that it isn't great, but it would consume all our resources to implement and maintain, and would not be much benefit for a project of this size.)

**Comment by threepointone** [comment]
I've had very bad experiences with nx myself.

**Comment by JacobMGEvans** [comment]
I am a big fan of turborepo.

**Comment by JacobMGEvans** [comment]
We should revisit this conversation.

**Comment by threepointone** [comment]
I noticed that our node_modules caching in CI doesn't even work properly, and we're reinstalling all npm deps every single time. Let's fix that first.

**Comment by petebacondarwin** [comment]
@JacobMGEvans - if you would like to move this forward, could you set out the specific benefits of how turborepo would speed up our builds? Right now, we only really have one significant package (the `wrangler` package) which takes very little time to build. I am not sure where we would improve the speed here?

**Reply by JacobMGEvans** [reply]
> @JacobMGEvans - if you would like to move this forward, could you set out the specific benefits of how turborepo would speed up our builds? Right now, we only really have one significant package (the `wrangler` package) which takes very little time to build. I am not sure where we would improve the speed here?

When I get back I will follow up with this, since it's moved to discussions

**Comment by JacobMGEvans** [comment]
I can write up something. To clarify I don't think it would have any major initial benefits now, I think it should be implemented now before the monorepo grows and we are maintaining/creating our own custom setup. In the long run I think it will have major benefits which again I will outline in the write-up; I will share the document here.
